refactor(ivy): make style application happen automatically

matsko opened the pull request:
This patch removes the need to issue a call to `stylingApply()`
at the end of change detection within a template or host bindings
function. What will happen instead is when the element is exited
(i.e. when `select(n)` is called, when a template exits or when
all host bindings for an element are processed) then styling will
be flushed automatically.

matsko commented:
Presubmit = http://test/OCL:261773025:BASE:261773055:1565041292941:b8525b22

Ivy Presubmit = http://test/OCL:261773025:BASE:261773450:1565041392803:86ef56b

mhevery commented:
Global approve

AndrewKushnir commented:
@matsko it looks like we have some regressions in Ivy presubmit (11 targets) and one of Guitar tests are "red" - that would trigger TAP failures for one of g3 apps. I'm adding "blocked" label for now, so we do not merge this PR before additional investigation of failed g3 targets. Thank you.

matsko committed:
refactor(ivy): make style application happen automatically

This patch removes the need to issue a call to `stylingApply()`
at the end of change detection within a template or host bindings
function. What will happen instead is when the element is exited
(i.e. when `select(n)` is called, when a template exits or when
all host bindings for an element are processed) then styling will
be flushed automatically.

matsko commented:
# master
node --no-turbo-inlining   33.41s user 0.27s system 102% cpu 32.962 total
node --no-turbo-inlining   33.22s user 0.23s system 102% cpu 32.727 total

# PR
node --no-turbo-inlining   49.63s user 0.36s system 102% cpu 48.639 total
node --no-turbo-inlining   50.00s user 0.38s system 102% cpu 49.022 total

# PR (without bit checks)
node --no-turbo-inlining   49.17s user 0.39s system 102% cpu 48.205 total

matsko commented:
Closed in favor of: https://github.com/angular/angular/pull/32259
